updates to support harbour, chum and SFOS 4.4 Sailjail requirements.

poetaster opened the pull request:
Updated spec to support chum, to make it into harbour store.
Updated desktop to satisfy Saijail needs, get into harbour, run on sfos 4.4

poetaster committed:
Updated spec to support chum, to make it into harbour store.
Updated desktop to satisfy Saijail needs, get into harbour, run on sfos
4.4

poetaster commented:
Builds are fine and tests on armv7 SFOS 3.4, and aarch64 SFOS 4.4 show everything BUT the shaders working well.

https://api.merproject.org/package/show/home:poetaster:kimmoli/paint

poetaster committed:
Added license to spec, url.

poetaster committed:
Added links for packaging repo issues que.

poetaster committed:
Added 0.9.1 tag.

poetaster committed:
Test modifications that will dissapear.

poetaster committed:
Fixed chum missing screenshot
Fixed a number opengl shaders missing precision or using fload syntax
Added some debug output.
Minor version bump.

poetaster committed:
Added some more filters for testing.
On image insert, the image Red and Blue channel are swapped. Now you can
swap em back:)

poetaster committed:
Added color channel swapping in all rgb direction
Added crosshatch, mirror, compound eye shaders.
Minor version bump.

poetaster committed:
Initial addition of undo functions.
These are cribbed from the naive methods in 'scribble'.

poetaster committed:
Minor change to support undo.
Undo is still PRIMITIVE and does not take layers into account.

kimmoli commented:
Nice to see contributions here. Thanks. Please let me know when everything works... I don't have energy to work with this at the moment...

poetaster commented:
> Nice to see contributions here. Thanks. Please let me know when everything works... I don't have energy to work with this at the moment...

This is all working code. The 'undo' functionality doesn't take layers into account 'properly' yet, but does no harm. The releases I've made are making the rounds. @attah, has found a fix for ImageProvider channel swapping which I'll integrate soon. 

Everything works, but needs polish. I'd merge now to keep the drift from becomming too great. So far I've:
* added shaders
* added the undo button component
* modified veriy little c++
* made QML modifications to support undo.

poetaster commented:
Question, did you have Paint in harbour? It'd be nice if it was there!

poetaster committed:
Implemented a colorspace conversion in the ImageProvider to stop the
image from going tinted.

poetaster committed:
reduced it to a more concise form.

kimmoli commented:
> Question, did you have Paint in harbour? It'd be nice if it was there!

Yes it was in Harbour

poetaster commented:
> > Question, did you have Paint in harbour? It'd be nice if it was there!
> 
> Yes it was in Harbour

I'll attach the last builds, which include the rgb swap fix in a prerelease in my repo. You're welcome to use them, the aarch64 users would appreciate it. If you wish, you can point people to my repo for support since I'm actively on it.

kimmoli commented:
Ok, i'll check at some point, might be on the weekend.
IIRC there was some issue with the toolbar scaling on some devices (or it was just my WIP toolbar improvements... that was last thing i touched before i abandoned these developments)

ping me on IRC on fri/sat ?

poetaster commented:
> Ok, i'll check at some point, might be on the weekend. IIRC there was some issue with the toolbar scaling on some devices (or it was just my WIP toolbar improvements... that was last thing i touched before i abandoned these developments)

Ok, I'll have a look. On 3.4 the roation of the toolbar causes a freeze (on the Volla/GS290 phones at least). 

> ping me on IRC on fri/sat ?

I'll try. I may be a bit busy with childcare ;)
 I've also started working on a better undo (derived from one in Imageworks) ... but I think it's not urgent.

poetaster committed:
Add numbers to revert/undo. lots of todos.
Add some experimental shaders.
Imageprovider test.

poetaster committed:
Bumped the version in spec.
